### PR TITLE
Switch to the new Hawk test runner

### PR DESCRIPTION
**Pull request by camsaul**
I spun out our test runner into a separate library, https://github.com/metabase/hawk, so we can share it between Metabase core and Harbormaster. This PR switches Metabase to use THE TEST HAWK 🦅

### CODE REVIEW

File: deps.edn [hunk 1]
```diff
@@ -203,8 +203,8 @@
     cloverage/cloverage          {:mvn/version "1.2.4"}
     com.gfredericks/test.chuck   {:mvn/version "0.2.13"}                    ; generating strings from regexes (useful with malli)
     djblue/portal                {:mvn/version "0.35.0"}                    ; ui for inspecting values
-    eftest/eftest                {:mvn/version "0.6.0"}
     io.github.camsaul/humane-are {:mvn/version "1.0.2"}
+    io.github.metabase/hawk      {:sha "126daf1c0aca962b9ca3a847716e0ad8e7821580"}
```
Comment (qnkhuat) [inline deps.edn:207]: What's stopping us from publishing it to clojars? I could help set it up if that's the reason.

**Review by qnkhuat — APPROVED** [review]

**Review by metamben — APPROVED** [review]
👍

File: deps.edn (hunk 1 above)
Comment (camsaul) [reply deps.edn:207]: I don't really think it's worth the effort since if you're using Leiningen you can just use Eftest directly

**Comment by camsaul** [comment]
Need to address https://github.com/metabase/hawk/issues/3 before merging this

**Comment by camsaul** [comment]
Ok, https://github.com/metabase/hawk/pull/4 is in now so we can merge once this is green

**Pull request merged**